patches/minecraft/net/minecraft/client/audio/SoundHandler.java.patch

```diff
@@ -0,0 +1,40 @@
+--- ../src-base/minecraft/net/minecraft/client/audio/SoundHandler.java
++++ ../src-work/minecraft/net/minecraft/client/audio/SoundHandler.java
+@@ -67,6 +67,7 @@
+     {
+         this.field_147697_e.func_148763_c();
+ 
++        Map<ResourceLocation, SoundList> resources = new java.util.IdentityHashMap<>();
+         for (String s : p_110549_1_.func_135055_a())
+         {
+             try
+@@ -79,6 +80,8 @@
+ 
+                         for (Entry<String, SoundList> entry : map.entrySet())
+                         {
++                            resources.put(new ResourceLocation(s, entry.getKey()), entry.getValue());
```

Comment:
Why do you don't simply push the progress bar where you create the map right now and set it in the vanilla loop and pop it where you do the loading now? So you don't need the new map at all

Reply:
I need to put all entries into the map here so that I can know the progress bar maximum when I construct it.

Reply:
Makes sense, didn't think about that